infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844Utils.java

```diff
@@ -70,27 +70,43 @@ public static TrustedSetup parseTrustedSetupFile(final String filePath) throws I
       // Number of G2 points
       final int g2Size = Integer.parseInt(reader.readLine());
       // List of G1 points, one on each new line
-      final List<Bytes48> g1Points = new ArrayList<>();
+      final List<Bytes> g1Points = new ArrayList<>();
       for (int i = 0; i < g1Size; i++) {
-        final Bytes48 g1Point = Bytes48.fromHexString(reader.readLine());
+        final Bytes g1Point = Bytes.fromHexString(reader.readLine(), CKZG4844.G1_POINT_SIZE);
         g1Points.add(g1Point);
       }
       // List of G2 points, one on each new line
       final List<Bytes> g2Points = new ArrayList<>();
       for (int i = 0; i < g2Size; i++) {
-        final Bytes g2Point = Bytes.fromHexString(reader.readLine(), G2_POINT_SIZE);
+        final Bytes g2Point = Bytes.fromHexString(reader.readLine(), CKZG4844.G2_POINT_SIZE);
         g2Points.add(g2Point);
       }
 
       return new TrustedSetup(g1Points, g2Points);
-    } catch (Exception ex) {
+    } catch (final Exception ex) {
       throw new IOException(String.format("Failed to parse trusted setup file\n: %s", filePath));
     }
   }
 
-  private static byte[] flattenBytes(final Stream<? extends Bytes> bytes, final int capacity) {
-    final ByteBuffer buffer = ByteBuffer.allocate(capacity);
-    bytes.map(Bytes::toArray).forEach(buffer::put);
-    return buffer.array();
+  private static byte[] flattenBytes(final List<Bytes> toFlatten, final int expectedSize) {
+    return flattenBytes(toFlatten, Function.identity(), expectedSize);
+  }
+
+  private static <T> byte[] flattenBytes(
+      final List<T> toFlatten, final Function<T, Bytes> bytesConverter, final int expectedSize) {
+    final byte[] flattened = new byte[expectedSize];
```

Comment:
Should we catch the potential `OutOfMemoryError` here and throw an `IllegalArgumentException` instead?

```java
try {
    final byte[] flattened = new byte[expectedSize];
    // ... rest of the code ...
} catch (OutOfMemoryError e) {
    throw new IllegalArgumentException("expectedSize is too big");
}
```

Reply:
I am not a fan of catching OOM exceptions. I added though a max size of 100 million bytes (100MB) which should be a decent default.

Reply:
So this isn't enough to batch verify a full BlobSidecarsByRange request at once.

* Right now, `MAX_REQUEST_BLOB_SIDECARS` is 768.
* A blob is 128KiB (4096 * 32) or 131072 bytes.
* 131072 * 768 is 100_663_296, which is just a hair more than 100MB.

The BlobSidecarsByRange message handler doesn't do this verification, but I think it's a reasonable limit.

Reply:
I think it's more nice default to be honest in any case. Changed it to:

```
private static final int MAX_BYTES_TO_FLATTEN = 100_663_296; // ~100.66 MB or 768 blobs
```

Reply:
Okay, thanks 😄